docs/reference/content/getting-started/async-quick-tour.md

```diff
@@ -0,0 +1,455 @@
+        System.out.println("Inserted!");
+    }
+});
+```
+
+Once the document has been inserted the `onResult` callback will be called and it will
+print "Inserted!".  Remember, in a normal application you would always check for the presence of
+errors in the `t` variable.
+
+## Add Multiple Documents
+
+To add multiple documents, you can use the `insertMany()` method.
+
+The following example will add multiple documents of the form:
+
+```javascript
+{ "i" : value }
+```
+
+Create the documents in a loop.
+
+```java
+List<Document> documents = new ArrayList<Document>();
+for (int i = 0; i < 100; i++) {
+    documents.add(new Document("i", i));
+}
+```
+
+To insert these documents to the collection, pass the list of documents to the
+`insertMany()` method.
+
+```java
+collection.insertMany(documents, new SingleResultCallback<Void>() {
+    @Override
+    public void onResult(final Void result, final Throwable t) {
+        System.out.println("Documents inserted!");
+    }
+});
+```
+
+## Count Documents in A Collection
+
+Now that we've inserted 101 documents (the 100 we did in the loop, plus
+the first one), we can check to see if we have them all using the
+[count()]({{< apiref "com/mongodb/client/MongoCollection#count--">}})
+method. The following code should print `101`.
+
+```java
+collection.count(
+  new SingleResultCallback<Long>() {
+      @Override
+      public void onResult(final Long count, final Throwable t) {
+          System.out.println(count);
+      }
+  });
+```
+
+## Query the Collection
+
+Use the
+[find()]({{< apiref "com/mongodb/client/MongoCollection.html#find--">}})
+method to query the collection.
+
+### Find the First Document in a Collection
+
+call the first() method on the result of the find() of method
+
+To get the first document in the collection, call the
+[first()]({{< apiref "com/mongodb/client/MongoIterable.html#first--">}})
+method on the [find()]({{< apiref "com/mongodb/client/MongoCollection.html#find--">}})
+operation. `collection.find().first()` returns the first document or null rather than a cursor.
+This is useful for queries that should only match a single document, or if you are
+interested in the first document only.
+
+{{% note %}}
+Sometimes you will need the same or similar callbacks more than once.  In these situations
+it makes sense to DRY (Do not Repeat Yourself) up your code and save the callback either
+as a concrete class or assign to a variable as below:
+
+```java
+SingleResultCallback<Document> printDocument = new SingleResultCallback<Document>() {
+    @Override
+    public void onResult(final Document document, final Throwable t) {
+        System.out.println(document.toJson());
+    }
+};
+```
+{{% /note %}}
+
+The following example passes the `printDocument` callback  to the `first` method:
+
+```java
+collection.find().first(printDocument);
+```
+
+
+The example should return immediately and eventually print the following document:
```

Comment:
This is a strange place to first point out the async nature of the methods that take callbacks, as it applies equally to all the methods discussed above as well.